File: 2021/06/6.jl
function solve(days, counts)
    for day in 1:days
        first = popfirst!(counts)
        push!(counts, 0)
        counts[7] += first
        counts[9] += first
    end
    return sum(counts)
end

nums = (x -> parse(Int, x)).(split(readline(), ','))
counts = [sum(nums .== i) for i in 0:9]

println(solve(80, copy(counts)))
println(solve(256, copy(counts)))
    

File: 2021/06/6.py
from collections import Counter

def solve(days, counter):
    for day in range(days):
        first, *rest = counter
        counter = rest + [0]
        counter[6] += first
        counter[8] += first
    return sum(counter)

inp = Counter(map(int, input().split(',')))
counts = [inp.get(i, 0) for i in range(10)]

print(solve(80, counts[:]))
print(solve(256, counts[:]))
    
